File: packages/lynx/host/ios/OpenChamberLynxBridge.swift
import Foundation

/// Page ↔ host channel. Mirrors `packages/lynx/src/host/bridge.ts`.
/// Native never owns the Lynx route stack — tab taps emit events; Lynx commits.
enum OpenChamberLynxBridgeEvent {
  case tabSelected(String)
  case backProgress(Double)
  case backCommit
  case backCancel
  case keyboardInset(Double)
  case predictiveBackProgress(Double)
  case predictiveBackCommit
  case predictiveBackCancel
}

enum OpenChamberLynxPageCommand {
  case setActiveTab(String)
  case hideHostTabChrome
  case showHostTabChrome
  case reportOccupancy(collapsedComposerHeight: Double)
  case openInstances
}

protocol OpenChamberLynxBridgeSink: AnyObject {
  func emit(_ event: OpenChamberLynxBridgeEvent)
}

/// Host-side dispatcher. Wire LynxView NativeModules / CustomEvent here once SDK links.
final class OpenChamberLynxBridge {
  weak var sink: OpenChamberLynxBridgeSink?

  func dispatch(_ command: OpenChamberLynxPageCommand) {
    // Host reacts (hide tab chrome, open instances chrome, etc.).
    // Intentionally no-op until Lynx SDK is linked — keeps compile surface honest.
    _ = command
  }

  func emit(_ event: OpenChamberLynxBridgeEvent) {
    sink?.emit(event)
  }
}

File: packages/lynx/host/android/app/src/main/java/com/yee94/openchamber/lynx/OpenChamberLynxBridge.kt
package com.yee94.openchamber.lynx

/** Mirrors packages/lynx/src/host/bridge.ts — native never owns the Lynx stack. */
sealed class LynxHostBridgeEvent {
    data class TabSelected(val tab: String) : LynxHostBridgeEvent()
    data class BackProgress(val progress: Double) : LynxHostBridgeEvent()
    object BackCommit : LynxHostBridgeEvent()
    object BackCancel : LynxHostBridgeEvent()
    data class KeyboardInset(val height: Double) : LynxHostBridgeEvent()
    data class PredictiveBackProgress(val progress: Double) : LynxHostBridgeEvent()
    object PredictiveBackCommit : LynxHostBridgeEvent()
    object PredictiveBackCancel : LynxHostBridgeEvent()
}

sealed class LynxPageToHostCommand {
    data class SetActiveTab(val tab: String) : LynxPageToHostCommand()
    object HideHostTabChrome : LynxPageToHostCommand()
    object ShowHostTabChrome : LynxPageToHostCommand()
    data class ReportOccupancy(val collapsedComposerHeight: Double) : LynxPageToHostCommand()
    object OpenInstances : LynxPageToHostCommand()
}

class OpenChamberLynxBridge {
    fun dispatch(command: LynxPageToHostCommand) {
        // Host reacts once LynxView is linked.
        @Suppress("UNUSED_EXPRESSION")
        command
    }
}
